wandb/integration/keras_core/metrics_logger.py

```diff
@@ -0,0 +1,155 @@
+    KerasCore.
+
+    This callback automatically logs the following to a W&B run page:
+    * system (CPU/GPU/TPU) metrics,
+    * train and validation metrics defined in `model.compile`,
+    * learning rate (both for a fixed value or a learning rate scheduler)
+
+    Notes:
+    If you resume training by passing `initial_epoch` to `model.fit` and you are using a
+    learning rate scheduler, make sure to pass `initial_global_step` to
+    `WandbMetricsLogger`. The `initial_global_step` is `step_size * initial_step`, where
+    `step_size` is number of training steps per epoch. `step_size` can be calculated as
+    the product of the cardinality of the training dataset and the batch size.
+
+    Arguments:
+        log_freq: ("epoch", "batch", or int) if "epoch", logs metrics
+            at the end of each epoch. If "batch", logs metrics at the end
+            of each batch. If an integer, logs metrics at the end of that
+            many batches. Defaults to "epoch".
+        initial_global_step: (int) Use this argument to correcly log the
+            learning rate when you resume training from some `initial_epoch`,
+            and a learning rate scheduler is used. This can be computed as
+            `step_size * initial_step`. Defaults to 0.
+    """
+
+    def __init__(
+        self,
+        log_freq: Union[LogStrategy, int] = "epoch",
+        initial_global_step: int = 0,
+        *args: Any,
+        **kwargs: Any,
+    ) -> None:
+        super().__init__(*args, **kwargs)
+
+        if wandb.run is None:
+            raise wandb.Error(
+                "You must call `wandb.init()` before WandbMetricsLogger()"
+            )
+
+        with telemetry.context(run=wandb.run) as tel:
+            tel.feature.keras_metrics_logger = True
+
+        if log_freq == "batch":
+            log_freq = 1
+
+        self.logging_batch_wise = isinstance(log_freq, int)
+        self.log_freq: Any = log_freq if self.logging_batch_wise else None
+        self.global_batch = 0
+        self.global_step = initial_global_step
+
+        if self.logging_batch_wise:
+            # define custom x-axis for batch logging.
+            wandb.define_metric("batch/batch_step")
+            # set all batch metrics to be logged against batch_step.
+            wandb.define_metric("batch/*", step_metric="batch/batch_step")
+        else:
+            # define custom x-axis for epoch-wise logging.
+            wandb.define_metric("epoch/epoch")
+            # set all epoch-wise metrics to be logged against epoch.
+            wandb.define_metric("epoch/*", step_metric="epoch/epoch")
+
+    def _get_lr(self) -> Union[float, None]:
+        try:
+            if isinstance(self.model.optimizer, keras.optimizers.Optimizer):
+                return float(self.model.optimizer.learning_rate.numpy().item())
+        except Exception:
+            if tf_backend_available:
+                if isinstance(self.model.optimizer.learning_rate, tf.Tensor):
```

Comment:
qq: you are importing TF inside an if-elif-else clause. Are they available to this method?

Reply:
That's the reason the flag `tf_backend_available` exists, to check beforehand if TF backend is available or not.